src/libpmemobj/backend_persistent.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2015, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * backend_persistent.h -- internal definitions for persistent backend
+ */
+
+typedef void (*persist_func)(void *addr, size_t len);
+
+#define	PERSISTENT_BACKEND_MAJOR 1
+#define	PERSISTENT_BACKEND_MINOR 0
+
+#define	MAX_INFO_SLOT 1024
+#define	MAX_CHUNK 10000L
+#define	CHUNKSIZE (1024 * 256)
+#define	POOL_SIGNATURE_LEN 16
+#define	POOL_SIGNATURE "MEMORY_POOL_HDR\0"
+#define	CHUNK_HEADER_MAGIC 0xC3F0
+#define	ZONE_MIN_SIZE (32 * (CHUNKSIZE))
+#define	INFO_SLOT_DATA_SIZE 28
+
+enum pool_flag {
+	POOL_FLAG_CLEAR_RECYCLED	=	0x0001,
+	POOL_FLAG_FILL_RECYCLED		=	0x0002,
+	POOL_FLAG_RUNTIME_TREE		=	0x0004,
+	POOL_FLAG_LAZY_LOAD		=	0x0008,
+};
+
+enum pool_state {
+	POOL_STATE_UNKNOWN,
+	POOL_STATE_OPEN,
+	POOL_STATE_CLOSED,
```

Comment:
It's always good to note that in enum definitions if someone needs to add a new state, in this case, they need to add it before MAX_POOL_STATE. Same for other enum definitions.